core/src/main/java/org/apache/iceberg/IncrementalFileCleanup.java

```diff
@@ -260,10 +259,14 @@ public void cleanFiles(TableMetadata beforeExpiration, TableMetadata afterExpira
         findFilesToDelete(manifestsToScan, manifestsToRevert, validIds, afterExpiration);
 
     deleteFiles(filesToDelete, "data");
-    LOG.warn("Manifests to delete: {}", Joiner.on(", ").join(manifestsToDelete));
-    LOG.warn("Manifests Lists to delete: {}", Joiner.on(", ").join(manifestListsToDelete));
     deleteFiles(manifestsToDelete, "manifest");
     deleteFiles(manifestListsToDelete, "manifest list");
+
+    if (!beforeExpiration.statisticsFiles().isEmpty()) {
+      Set<String> expiredStatisticsFilesLocations =
+          expiredStatisticsFilesLocations(beforeExpiration, afterExpiration);
+      deleteFiles(expiredStatisticsFilesLocations, "statistics files");
+    }
```

Comment:
Nit: I think we could just have expireStatisticsFilesLocation short circuit before computing  in case before expiration is empty. Rather than have the if(!beforeExpiration.statisticsFiles().isEmpty()) in both cleanup implementations